### PR TITLE
[MINOR][CORE][DOCS] Fix inconsistent description of showConsoleProgress

### PR DESCRIPTION
**Pull request by beliefer**
## What changes were proposed in this pull request?

The latest docs http://spark.apache.org/docs/latest/configuration.html contains some description as below:

spark.ui.showConsoleProgress | true | Show the progress bar in the console. The progress bar shows the progress of stages that run for longer than 500ms. If multiple stages run at the same time, multiple progress bars will be displayed on the same line.
-- | -- | --

But the class `org.apache.spark.internal.config.UI` define the config `spark.ui.showConsoleProgress` as below:
```
val UI_SHOW_CONSOLE_PROGRESS = ConfigBuilder("spark.ui.showConsoleProgress")
    .doc("When true, show the progress bar in the console.")
    .booleanConf
    .createWithDefault(false)
```
So I think there are exists some little mistake and lead to confuse reader.

## How was this patch tested?

No need UT.

### CODE REVIEW

**Comment by beliefer** [comment]
@srowen Could you take a look at this PR?

**Review by srowen — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

File: docs/configuration.md [hunk 1]
```diff
@@ -1062,7 +1062,7 @@ Apart from these, the following properties are also available, and may be useful
 </tr>
 <tr>
   <td><code>spark.ui.showConsoleProgress</code></td>
-  <td>true</td>
+  <td>false</td>
   <td>
     Show the progress bar in the console. The progress bar shows the progress of stages
```
Comment (HyukjinKwon) [inline docs/configuration.md]: Shall we mention that this is enabled in shells by default?

Comment (dongjoon-hyun) [reply docs/configuration.md]: +1 for @HyukjinKwon 's comment.

Comment (beliefer) [reply docs/configuration.md]: @HyukjinKwon @dongjoon-hyun I will add a mention about this.

File: docs/configuration.md [hunk 2]
```diff
@@ -1062,11 +1062,14 @@ Apart from these, the following properties are also available, and may be useful
...
     that run for longer than 500ms. If multiple stages run at the same time, multiple
     progress bars will be displayed on the same line.
+    <br/>
+    <em>Note:</em> In non-shell environment, the default value of spark.ui.showConsoleProgress is true or
```
Comment (HyukjinKwon) [inline docs/configuration.md]: I think it's true in shells by default (not non-shell environment)

Comment (beliefer) [reply docs/configuration.md]: Sorry, I make a mistake.

File: docs/configuration.md [hunk 3]
```diff
@@ -1062,11 +1062,14 @@ Apart from these, the following properties are also available, and may be useful
...
   <td><code>spark.ui.showConsoleProgress</code></td>
-  <td>true</td>
+  <td>false</td>
   <td>
     Show the progress bar in the console. The progress bar shows the progress of stages
     that run for longer than 500ms. If multiple stages run at the same time, multiple
     progress bars will be displayed on the same line.
+    <br/>
+    <em>Note:</em> In shell environment, the default value of spark.ui.showConsoleProgress is true or
+    specified by user.
```
Comment (HyukjinKwon) [inline docs/configuration.md]: Let's remove the last `or specified by user.`. Sounds confusing. LGTM otherwise.

Comment (beliefer) [reply docs/configuration.md]: OK. I will remove it.

**Comment by HyukjinKwon** [comment]
Merged to master, branch-2.4 and branch-2.3.

**Comment by beliefer** [comment]
@HyukjinKwon @srowen @dongjoon-hyun Thanks for all your help!

**Pull request closed without merging**